stdlib/jdbc/src/main/java/org/ballerinax/jdbc/statement/BatchUpdateStatement.java

```diff
@@ -152,6 +152,9 @@ public BatchUpdateStatement(ObjectValue client, SQLDatasource datasource, String
     // If such other databases are identified they can be included here.
     // The name of the database is being checked because there is no way to identify through the API.
     private boolean isGeneratedKeyReturningSupported() {
+        if (!datasource.supportsGetGeneratedKeys()) {
+            return false;
+        }
         return !Constants.DatabaseNames.ORACLE.equals(datasource.getDatabaseProductName())
```

Comment:
Is there a possibility to remove the Check for Oracle and MSSQL server check below? As per the comment `The name of the database is being checked because there is no way to identify through the API.` it seems it was done without knowing the `Connection.getMetaData().supportsGetGeneratedKeys()` API method.

Reply:
No. "supportsGetGeneratedKeys" - function if true guarantees at least insert operation is supported. Others are vendor-specific. Oracle official driver returns true for this API but Oracle do not support key generation in the batch operations.

Reply:
Docs for supportsGetGeneratedKeys
https://docs.oracle.com/javase/8/docs/api/java/sql/DatabaseMetaData.html?is-external=true#supportsGetGeneratedKeys--